fix: Fix more browser tests

RoboErikG opened the pull request:
<!--
  - Thanks for submitting code to Blockly!  Please fill out the following as part of
  - your pull request so we can review your code more easily.
  -->

## The basics

This fixes the remaining browser tests

- [x] I [validated my changes](https://developers.google.com/blockly/guides/contribute/core#making_and_verifying_a_change)

## The details
### Resolves

<!-- TODO: What Github issue does this resolve? Please include a link. -->
Fixes [#8953](https://github.com/google/blockly/issues/8953)

### Proposed Changes

Lots of fixes to the browser tests

### Reason for Changes

<!--TODO: Explain why these changes should be made.  Include screenshots if applicable. -->

### Test Coverage

<!-- TODO: Please create unit tests, and explain here how they cover
           your changes, or tell us how you tested it manually. If
           your changes include browser-specific behaviour, include
           information about the browser and device that you used for
           testing. -->

### Documentation

<!-- TODO: Does any documentation need to be created or updated because of this PR?
  -        If so please explain.
  -->

### Additional Information

<!-- Anything else we should know? -->

RoboErikG committed:
Work on fixing more browser tests

RoboErikG committed:
Remove .only

RoboErikG committed:
Fix tests for opening categories

RoboErikG committed:
Fix procedure tests

RoboErikG committed:
Disable test to drag all blocks out and fix comment resize test

RoboErikG committed:
Merge branch 'google:develop' into fix-browser-tests-2025-06

gonfunko commented on tests/browser/test/delete_blocks_test.mjs:
```diff
@@ -194,7 +194,8 @@ suite('Delete blocks', function (done) {
   });
 
-  test('Redo block deletion', async function () {
+  // TODO(#9029) enable this test once deleting a block doesn't lose focus
```
This was fixed in https://github.com/google/blockly/pull/9195, can this be reenabled now?

RoboErikG replied:
Done.

gonfunko commented on tests/browser/test/test_setup.mjs:
```diff
@@ -295,6 +298,55 @@ export async function getBlockTypeFromCategory(
+        .getBlocksByType(blockType)[0];
+      if (!block.isCollapsed()) {
+        for (const input of block.inputList) {
+          for (const field of input.fieldRow) {
```
There's a recently-ish introduced `getFields()` that returns a generator that might be more convenient here, but not a blocker.

RoboErikG replied:
Done.

gonfunko commented on tests/browser/test/test_setup.mjs:
```diff
@@ -448,6 +500,9 @@ export async function switchRTL(browser) {
   const flyoutBlock = await getNthBlockOfCategory(browser, categoryName, n);
+  while (!(await elementInBounds(browser, flyoutBlock))) {
+    await scrollFlyout(browser, 0, 50);
+  }
```
Could you use `WorkspaceSvg.scrollBoundsIntoView(flyoutBlock.getBoundingRectangleWithoutChildren())` rather than scrolling repeatedly by a fixed amount until it's on screen?

gonfunko replied:
This would seemingly allow you to get rid of `elementInBounds`, since it should be safe to do unconditionally.

RoboErikG replied:
It took some reworking of a few of the test methods, but does seem to be cleaner and work better overall. =)

gonfunko commented on tests/browser/test/test_setup.mjs:
```diff
@@ -474,15 +529,44 @@ export async function dragBlockTypeFromFlyout(
   );
+  while (!(await elementInBounds(browser, flyoutBlock))) {
+    await scrollFlyout(browser, 0, 50);
+  }
```
Here as well

gonfunko commented on tests/browser/test/toolbox_drag_test.mjs:
```diff
@@ -57,28 +57,29 @@ const testCategories = [
-    return vertInView && horInView;
-  }, element);
+  // Unicode escape to close flyout.
+  await browser.keys(['\uE00C']);
```
Can this be `webdriverio.Key.Escape`?

RoboErikG replied:
Done here and elsewhere.

RoboErikG committed:
Re-enable undo/redo tests now that focus is working

RoboErikG committed:
Responses to comments

- Switch to using scrollBoundsIntoView instead of scrolling the flyout
- Use webdriverio Key.Escape instead of the string code for it

RoboErikG reviewed (commented):
Updated based on comments

RoboErikG committed:
Fix formatting

RoboErikG committed:
Add a todo for enabling the toolbox categories tests
